packages/react-reconciler/src/ReactFiberCompleteWork.js

```diff
@@ -884,6 +885,13 @@ function completeWork(
           workInProgress.effectTag |= Update;
         }
       }
+      if (
+        enableSuspenseCallback &&
+        workInProgress.memoizedProps.suspenseCallback != null
+      ) {
+        // Always notify the callback
+        workInProgress.effectTag |= Update;
```

Comment:
Since this will only be called (for this purpose) if updateQueue is non-null, you should check `if (workInProgress.updateQueue !== null)` before adding this effect.